app/routes/components/timer/Timer.js

```diff
@@ -7,25 +7,28 @@ import styles from "./styles";
 
 class Timer extends Component {
   static propTypes = {
-    startingDate: PropTypes.string.isRequired
+    date: PropTypes.string.isRequired,
+    timerStyle: View.propTypes.style.isRequired
   };
 
   static msToTime(duration) {
     let seconds = parseInt(duration / 1000 % 60, 10);
     let minutes = parseInt(duration / (1000 * 60) % 60, 10);
     let hours = parseInt(duration / (1000 * 60 * 60) % 24, 10);
+    let days = parseInt(duration / (1000 * 60 * 60 * 24) % 365, 10);
 
+    days = days < 10 ? `0${days}` : days;
     hours = hours < 10 ? `0${hours}` : hours;
     minutes = minutes < 10 ? `0${minutes}` : minutes;
     seconds = seconds < 10 ? `0${seconds}` : seconds;
 
-    return `${hours}:${minutes}:${seconds}`;
+    return `${days}:${hours}:${minutes}:${seconds}`;
   }
 
   constructor(props) {
     super(props);
     this.state = {
-      startingTime: new Date(this.props.startingDate),
+      date: new Date(this.props.date),
       elapsedTime: 0
```

Comment:
the way you did it make it really hard to test (-> u have to use a mock of Date)
what you realy have is a component that display the diff of 2 Date
then on of those date is updated every X millisecond -> that part is an hoc